kedro_azureml/datasets/folder_dataset.py

```diff
@@ -0,0 +1,96 @@
+import logging
+from copy import deepcopy
+from typing import Any, Dict, Type, Union
+from warnings import warn
+
+from kedro.io.core import (
+    VERSION_KEY,
+    VERSIONED_FLAG_KEY,
+    AbstractDataSet,
+    DataSetError,
+    parse_dataset_definition,
+)
+
+from kedro_azureml.distributed.utils import is_distributed_master_node
+
+logger = logging.getLogger(__name__)
+
+
+class AzureMLFolderDataset(AbstractDataSet):
+    def __init__(
+        self,
+        path: str,
+        dataset: Union[str, Type[AbstractDataSet], Dict[str, Any]],
+        filepath_arg: str = "filepath",
+    ):
+        """Creates a new instance of ``AzureMLFolderDataset``.
```

Comment:
You can use autodocs to mention in in the documentation, follow the example: https://github.com/getindata/kedro-azureml/blob/c220ac31c062d4cceda17b6fcec7053178d47217/docs/source/04_data_assets.rst